docs/micro-rdk/board/esp32.md

```diff
@@ -0,0 +1,122 @@
+---
+title: "Configure an ESP32 board"
+linkTitle: "esp32"
+weight: 20
+type: "docs"
+description: "Configure an esp32 board."
+images: ["/icons/components/board.svg"]
+tags: ["board", "components"]
+# SMEs: Gautham, Rand
+---
+
+{{% alert title="REQUIREMENTS" color="caution" %}}
+
+Follow the [setup guide](/installation/prepare/microcontrollers/) to prepare your ESP32 for running micro-RDK before configuring a `esp32` board.
+
+{{% /alert %}}
+
+To add an `esp32` board, navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com) and select **Raw JSON** mode.
+
+Copy the following JSON template and paste it into your configuration inside the `"components"` array:
+
+{{< tabs name="Configure an esp32 Board" >}}
+{{% tab name="JSON Template"%}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "attributes": {
+    "pins": [
+      <int>
+    ],
+    "analogs": [
+      {
+        "pin": "<number>",
+        "name": "<your-analog-name>"
+      }
+    ]
+  },
+  "depends_on": [],
+  "model": "esp32",
+  "name": "<your-board-name>",
+  "type": "board"
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "attributes": {
+    "pins": [15],
+    "analogs": [
+      {
+        "pin": "34",
+        "name": "sensor"
+      }
+    ]
+  },
+  "depends_on": [],
+  "model": "esp32",
+  "name": "board",
+  "type": "board"
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Edit and fill in the attributes as applicable.
+Click **Save config**.
+
+The following attributes are available for `esp32` boards:
+
+<!-- prettier-ignore -->
+| Name | Type | Inclusion | Description |
+| ---- | ---- | --------- | ----------- |
+| `analogs` | object | Optional | Attributes of any pins that can be used as analog-to-digital converter (ADC) inputs. See [configuration info](#analogs). |
+| `i2cs` | object | Optional | Any Inter-Integrated Circuit (I<sup>2</sup>C) pins' bus index and name. See [configuration info](#i2cs). |
+| `digital_interrupts` | object | Optional | Any digital interrupts' GPIO number. See [configuration info](#digital_interrupts). |
+| `pins` | object | Required | The GPIO number of any GPIO pins you wish to use as input/output with the [`GPIOPin` API](/program/apis/#gpio-pins). |
+
+Any pin not specified in either `"pins"` or `"digital_interrupts"` cannot be interacted with through the [Board API](/components/board/#api).
+Interaction with digital interrupts is only supported with the [Board API](/components/board/#api), these digital interrupts cannot be used as software interrupts in driver implementations.
+
+{{< alert title="Info" color="info" >}}
```

Comment:
Consider moving this tip up above the JSON template/example tabs, since that's where users may be used to finding a builder mode tab

Reply:
+1

Reply:
moved!